llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -543,11 +543,10 @@ class WAWEdges : public ScheduleDAGMutation {
     LiveRegs.init(*TRI);
     bool AddReservedRegs = true;
     if (Scheduler) {
+      assert(!Scheduler->doMBBSchedRegionsTopDown());
```

Comment:
CHECK: this assert is specific for the prologue handling, and the code needs to be extended for epilogue handling.

Reply:
No, WAW handling could run on any region. Here I'm making sure that `BS.getBottom()` actually represents the bottom region.

Reply:
Perhaps move it to `getBottom()` then?